entgql/plugin.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package entgql
+
+import (
+	"strings"
+
+	"github.com/99designs/gqlgen/codegen/config"
+	"github.com/99designs/gqlgen/plugin"
+	"github.com/vektah/gqlparser/v2/ast"
+	"github.com/vektah/gqlparser/v2/formatter"
+)
+
+// CreatePlugin create the plugin for GQLGen
+func (e *Extension) CreatePlugin() plugin.Plugin {
+	return &gqlgenPlugin{
+		schema: e.schema,
+		models: e.models,
+	}
+}
+
+type gqlgenPlugin struct {
```

Comment:
Nice, but it's actually more complex than setting `autobind` or the `@goModel` directive in the generated schema.

Reply:
autobind didnt work in all cases. So, the user still need to manual config mapping. Then, we have to read config.yaml to recogine them.

If we can detect and mapping it correctly, we not need to read config.yaml anymore and not must to depends on gqlgen.